tests/core/mempool/test_mempool_manager.py

```diff
@@ -552,3 +575,43 @@ async def test_get_items_not_in_filter() -> None:
     sb2_and_3_filter = PyBIP158([bytearray(sb2_name), bytearray(sb3_name)])
     result = await mempool_manager.get_items_not_in_filter(sb2_and_3_filter)
     assert result == [mempool_item1]
+
+
+@pytest.mark.asyncio
+async def test_new_peak_remove_from_pool() -> None:
+    mempool_manager = await instantiate_mempool_manager(get_coin_record_for_test_coins)
+    # Create a bunch of mempool items that spend TEST_COIN in order to setup
+    # a mapping of that coin's ID to several spend bundles
+    (sb1, sb2, sb3) = (
+        spend_bundle_from_conditions([[ConditionOpcode.CREATE_COIN, IDENTITY_PUZZLE_HASH, a]], c)
+        for (a, c) in zip((1, 2, 3), (TEST_COIN, TEST_COIN2, TEST_COIN3))
+    )
+    sb12 = SpendBundle.aggregate([sb1, sb2])
+    sb12_name = sb12.name()
+    mi12 = mempool_item_from_spendbundle(sb12)
+    sb13 = SpendBundle.aggregate([sb1, sb3])
+    sb13_name = sb13.name()
+    mi13 = mempool_item_from_spendbundle(sb13)
+    sb123 = SpendBundle.aggregate([sb1, sb2, sb3])
+    sb123_name = sb123.name()
+    mi123 = mempool_item_from_spendbundle(sb123)
+    mempool_manager.mempool.removal_coin_id_to_spendbundle_ids = {
```

Comment:
this doesn't look OK. tests really should be using the public interface for the classes they test

Reply:
We'll add tests for this when we activate the 1 to n mapping of coin IDs to spend bundles.